include/itkAttenuationImageFilter.h

```diff
@@ -0,0 +1,204 @@
+
+  using InputImageType = TInputImage;
+  using InputImagePointer = typename TInputImage::Pointer;
+
+  using OutputImageType = TOutputImage;
+  using OutputImagePointer = typename TOutputImage::Pointer;
+
+  using MaskImageType = TMaskImage;
+  using MaskImagePointer = typename TMaskImage::Pointer;
+  using MaskPixelType = typename TMaskImage::PixelType;
+
+  using InputRegionType = typename TInputImage::RegionType;
+  using InputSizeType = typename TInputImage::SizeType;
+  using InputIndexType = typename TInputImage::IndexType;
+  using InputPixelType = typename TInputImage::PixelType;
+
+  using OutputRegionType = typename TOutputImage::RegionType;
+  using OutputPixelType = typename TOutputImage::PixelType;
+
+  itkSetInputMacro(MaskImage, TMaskImage);
+  itkGetInputMacro(MaskImage, TMaskImage);
+
+  itkGetConstMacro(LabelValue, MaskPixelType);
+  itkSetMacro(LabelValue, MaskPixelType);
+
+  itkGetConstMacro(Direction, unsigned int);
+  itkSetMacro(Direction, unsigned int);
+
+  itkGetConstMacro(SamplingFrequencyMHz, float);
+  itkSetMacro(SamplingFrequencyMHz, float);
+
+  itkGetConstMacro(FrequencyBandStartMHz, float);
+  itkSetMacro(FrequencyBandStartMHz, float);
+
+  itkGetConstMacro(FrequencyBandEndMHz, float);
+  itkSetMacro(FrequencyBandEndMHz, float);
+
+  itkGetConstMacro(ConsiderNegativeAttenuations, bool);
+  itkSetMacro(ConsiderNegativeAttenuations, bool);
+
+  itkGetConstMacro(PadPixelUpperBounds, unsigned int);
+  itkSetMacro(PadPixelUpperBounds, unsigned int);
+  itkGetConstMacro(PadPixelLowerBounds, unsigned int);
+  itkSetMacro(PadPixelLowerBounds, unsigned int);
+
+  itkGetConstMacro(PadSpatialUpperBoundsMM, float);
+  itkSetMacro(PadSpatialUpperBoundsMM, float);
+  itkGetConstMacro(PadSpatialLowerBoundsMM, float);
+  itkSetMacro(PadSpatialLowerBoundsMM, float);
+
+  // Alias for setting direction of RF waveform in data collection
+  void
+  SetScanDirection(unsigned int direction)
+  {
+    this->SetDirection(direction);
+  };
+  unsigned int
+  GetScanDirection()
+  {
+    return this->GetDirection();
+  };
+
+  void
+  PrintSelf(std::ostream & os, Indent indent) const override;
+
+protected:
+  AttenuationImageFilter();
+  ~AttenuationImageFilter() override = default;
+
+  /** Initialize some accumulators before any chunks are processes */
+  void
+  BeforeThreadedGenerateData() override;
+
+  /** Do final mean and variance computation from data accumulated in threads.
+   */
+  void
+  AfterThreadedGenerateData() override;
+
+  void
+  DynamicThreadedGenerateData(const OutputRegionType &) override;
+
+  const ImageRegionSplitterBase *
+  GetImageRegionSplitter() const override;
+
+  /** Compute attenuation between two pixels in the RF spectra vector image.
+   *  Assumes that image spacing is in MM. */
+  OutputPixelType
+  ComputeAttenuation(const InputIndexType & end, const InputIndexType & start) const;
+
+private:
+  /** Label value indicating pixel should be included in analysis */
```

Comment:
We should change this so that if `m_LabelValue==0`, then all non-zero pixels are considered as belonging to the mask (inclusion/organ/tumor). And if `m_LabelValue` is non-zero, then only the pixels with that exact value are considered. I think that most masks in ITK use maxValue by default (255 for uchar).

Reply:
> I think that most masks in ITK use maxValue by default (255 for uchar).

Thanks for the insight. I will make the change and add a comment in the header to document.